apps/007_ofsted_market_analysis/app.py

```diff
@@ -2,113 +2,498 @@
+        file.rename(columns = {'Latest overall Effectiveness grade from last full inspection':'Overall effectiveness'}, inplace = True)
+        file.rename(columns = {'Overall Effectiveness':'Overall effectiveness'}, inplace = True)
+        file.rename(columns = {'CYP Safety':'CYP safety'}, inplace = True)
+        file.rename(columns = {'Leadership and Management':'Leadership and management'}, inplace = True)
+        file.rename(columns = {'Max Users':'Number of registered places'}, inplace = True)
+        file.rename(columns = {'Emotional and Behavioural Difficulties':'Emotional and behavioural difficulties'}, inplace = True)
+        file.rename(columns = {'Mental Disorders':'Mental disorders'}, inplace = True)
+        file.rename(columns = {'Sensory Impairment':'Sensory impairment'}, inplace = True)
+        file.rename(columns = {'Present impairment':'Present alcohol problems'}, inplace = True)
+        file.rename(columns = {'Present Alcohol Problems':'Present alcohol problems'}, inplace = True)
+        file.rename(columns = {'Present Drug Problems':'Present drug problems'}, inplace = True)
+        file.rename(columns = {'Learning Difficulty':'Learning difficulty'}, inplace = True)
+        file.rename(columns = {'Physical Disabilities':'Physical disabilities'}, inplace = True)
+        
+        #st.dataframe(file)
+
+        # Put name of datasets and datasets themselves into a dictionary
+        files_dict[name] = file
+        
+      
+    #files_dict
 
-    # Take only the first 48 columns and rows where URN is not null
-    df = df.iloc[:, : 47]
-    df = df.dropna(subset=['URN'])
+    # Append all datasets together to check which columns line up
+    df = pd.concat(list(files_dict.values()), axis=0).reset_index()
 
-    # Filter out Sectors "Local Authority" and "Health Authority"
-    df = df[~df['Sector'].isin(['Local Authority', 'Health Authority'])]
+    # Match local authority to region
+    df = df.merge(regions,how='left',on='Local authority')
 
-    # Where owner fields are blank, replace with setting fields
-    df['Owner ID'][df['Owner ID'].isna()] = df['URN']
-    df['Owner name'][df['Owner name'].isna()] = df['Setting name']
+    # Select only necessary columns
+    df = df[['year',
+             'month',
+             'homestype',
+             'Local authority',
+             'Region_y',
+             'URN',
+             'Provider type',
+             'Provider subtype',
+             'Sector',
+             'Setting name',
+             'Registration status',
+             'Owner ID',
+             'Owner name',
+             'Overall effectiveness',
+             'CYP safety',
+             'Leadership and management',
+             'Number of registered places',
+             'Emotional and behavioural difficulties',
+             'Mental disorders',
+             'Sensory impairment',
+             'Present alcohol problems',
+             'Present drug problems',
+             'Learning difficulty',

```

Comment:
I think I've only ever done this with a dicitonary and lambda functions, this is really interesting that there's a pre made function